modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/servlet/taglib/ChangeTrackingIndicatorDynamicInclude.java

```diff
@@ -270,32 +261,6 @@ protected void activate(Map<String, Object> properties) {
 			CTConfiguration.class, properties);
 	}
 
-	private void _getConflictIconData(
-		long classNameId, long classPK, CTCollection currentCTCollection,
-		Map<String, Object> data, HttpServletRequest httpServletRequest,
-		ThemeDisplay themeDisplay) {
-
-		if (currentCTCollection == null) {
-			return;
-		}
-
-		ResourceURL getConflictInfoURL =
-			(ResourceURL)_portal.getControlPanelPortletURL(
-				httpServletRequest, themeDisplay.getScopeGroup(),
-				CTPortletKeys.PUBLICATIONS, 0, 0,
-				PortletRequest.RESOURCE_PHASE);
-
-		getConflictInfoURL.setParameter(
-			"classNameId", String.valueOf(classNameId));
-		getConflictInfoURL.setParameter("classPK", String.valueOf(classPK));
-		getConflictInfoURL.setParameter(
-			"currentCTCollectionId",
-			String.valueOf(currentCTCollection.getCtCollectionId()));
-		getConflictInfoURL.setResourceID("/change_tracking/get_conflict_info");
-
-		data.put("getConflictInfoURL", getConflictInfoURL.toString());
-	}
-
```

Comment:
Since there isn't a `currentCTCollection` variable available in `getTimelineData`, I used CTCollectionThreadLocal to obtain the current one and replaced the early return conditional to check if that the user is currently not on production.
```

if (!CTCollectionThreadLocal.isProductionMode()) {
	ResourceURL getConflictInfoURL =
		(ResourceURL)_portal.getControlPanelPortletURL(
			httpServletRequest, themeDisplay.getScopeGroup(),
			CTPortletKeys.PUBLICATIONS, 0, 0,
			PortletRequest.RESOURCE_PHASE);

	getConflictInfoURL.setParameter(
		"classNameId", String.valueOf(classNameId));
	getConflictInfoURL.setParameter("classPK", String.valueOf(classPK));
	getConflictInfoURL.setParameter(
		"currentCTCollectionId",
		String.valueOf(CTCollectionThreadLocal.getCTCollectionId()));
	getConflictInfoURL.setResourceID("/change_tracking/get_conflict_info");

	data.put("getConflictInfoURL", getConflictInfoURL.toString());
}

```

Reply:
The `getTimelineData` used to take in a ctCollection https://github.com/liferay/liferay-portal/blob/master/modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/servlet/taglib/ChangeTrackingIndicatorDynamicInclude.java#L613-L614, and the ctCollection is being retrieved from the CTPreferences https://github.com/liferay/liferay-portal/blob/master/modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/servlet/taglib/ChangeTrackingIndicatorDynamicInclude.java#L163-L172. Let's try to keep the old behavior to avoid causing regressions.